File: WordTarget/WordTargetCoreTests/LinkTypesTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace WordTargetCore
{
    [TestClass]
    public class LinkTypesTest
    {
        [TestMethod]
        public void CountLetters()
        {
            LinkTypes.LetterCounter letterCount = LinkTypes.CountLetters("system");
            Assert.AreEqual(2, letterCount.CountFor('s'));
            Assert.AreEqual(1, letterCount.CountFor('y'));
            Assert.AreEqual(1, letterCount.CountFor('t'));
            Assert.AreEqual(1, letterCount.CountFor('e'));
            Assert.AreEqual(1, letterCount.CountFor('m'));
            Assert.AreEqual(0, letterCount.CountFor('a'));
            Assert.AreEqual(0, letterCount.CountFor('b'));
        }
        [TestMethod]
        public void CompareLetterCount()
        {
            LinkTypes.LetterCounter letterCount1 = LinkTypes.CountLetters("system");
            LinkTypes.LetterCounter letterCount2 = LinkTypes.CountLetters("system");
            System.Console.WriteLine(letterCount1.ToString());
            Assert.AreEqual(letterCount1, letterCount2);

            letterCount1 = LinkTypes.CountLetters("system");
            letterCount2 = LinkTypes.CountLetters("sytem");
            System.Console.WriteLine(letterCount1.ToString());
            Assert.AreNotEqual(letterCount1, letterCount2);

            letterCount1 = LinkTypes.CountLetters("system");
            letterCount2 = LinkTypes.CountLetters("yetmss");
            System.Console.WriteLine(letterCount1.ToString());
            Assert.AreEqual(letterCount1, letterCount2);

            letterCount1 = LinkTypes.CountLetters("abc");
            letterCount2 = LinkTypes.CountLetters("cde");
            System.Console.WriteLine(letterCount1.ToString());
            Assert.AreNotEqual(letterCount1, letterCount2);

            letterCount1 = LinkTypes.CountLetters("abcd");
            letterCount2 = LinkTypes.CountLetters("aaaa");
            System.Console.WriteLine(letterCount1.ToString());
            Assert.AreNotEqual(letterCount1, letterCount2);
        }

        [TestMethod]
        public void IsAnagram()
        {
            Assert.IsTrue(LinkTypes.IsAnagram("bat", "tab"));
            Assert.IsTrue(LinkTypes.IsAnagram("angel", "angle"));
            Assert.IsFalse(LinkTypes.IsAnagram("cat", "angle"));
            Assert.IsFalse(LinkTypes.IsAnagram("cat", "cat"));
        }

        [TestMethod]
        public void IsOneLetterChange()
        {
            Assert.IsTrue(LinkTypes.IsOneLetterChange("moon", "moan"));
            Assert.IsTrue(LinkTypes.IsOneLetterChange("cart", "care"));
            Assert.IsFalse(LinkTypes.IsOneLetterChange("cat", "bit"));
            Assert.IsFalse(LinkTypes.IsOneLetterChange("cat", "angle"));
            Assert.IsFalse(LinkTypes.IsOneLetterChange("cat", "cat"));
        }

        [TestMethod]
        public void IsOneLetterAdd()
        {
            Assert.IsTrue(LinkTypes.IsOneLetterAdd("pan", "pain"));
            Assert.IsTrue(LinkTypes.IsOneLetterAdd("car", "cart"));
            Assert.IsFalse(LinkTypes.IsOneLetterAdd("cat", "bite"));
            Assert.IsFalse(LinkTypes.IsOneLetterAdd("raw", "straw"));
            Assert.IsFalse(LinkTypes.IsOneLetterAdd("cat", "cat"));
        }
        
        [TestMethod]
        public void IsOneLetterRemove()
        {
            Assert.IsTrue(LinkTypes.IsOneLetterRemove("pain", "pan"));
            Assert.IsTrue(LinkTypes.IsOneLetterRemove("cart", "car"));
            Assert.IsFalse(LinkTypes.IsOneLetterRemove("cat", "bite"));
            Assert.IsFalse(LinkTypes.IsOneLetterRemove("straw", "raw"));
            Assert.IsFalse(LinkTypes.IsOneLetterRemove("cat", "cat"));
        }


        [TestMethod]
        public void ToString1()
        {
            Assert.AreEqual("OneLetterChange", LinkType.OneLetterChange.ToString());
            Assert.AreEqual("OneLetterAddOrRemove", LinkType.OneLetterAddOrRemove.ToString());
            Assert.AreEqual("Anagram", LinkType.Anagram.ToString());
            Assert.AreEqual("Synonym", LinkType.Synonym.ToString());
            Assert.AreEqual("Antonym", LinkType.Antonym.ToString());
            Assert.AreEqual("WordAssociation", LinkType.WordAssociation.ToString());
        }

        [TestMethod]
        public void FromString1()
        {
            Assert.AreEqual(LinkType.OneLetterChange, LinkTypes.FromString("OneLetterChange"));
            Assert.AreEqual(LinkType.OneLetterAddOrRemove, LinkTypes.FromString("OneLetterAddOrRemove"));
            Assert.AreEqual(LinkType.Anagram, LinkTypes.FromString("Anagram"));
            Assert.AreEqual(LinkType.Synonym, LinkTypes.FromString("Synonym"));
            Assert.AreEqual(LinkType.Antonym, LinkTypes.FromString("Antonym"));
            Assert.AreEqual(LinkType.WordAssociation, LinkTypes.FromString("WordAssociation"));
        }

    }
}

File: word-target/word-target-core/src/test/java/it/carcassi/wordtarget/core/LinkTypesTest.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package it.carcassi.wordtarget.core;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import static it.carcassi.wordtarget.core.CommonTestObjects.*;

/**
 *
 * @author Matteo
 */
public class LinkTypesTest {

    @Test
    public void testIsAnagram() {
        assertTrue(LinkTypes.isAnagram("bat", "tab"));
        assertTrue(LinkTypes.isAnagram("angel", "angle"));
        assertFalse(LinkTypes.isAnagram("cat", "angle"));
        assertFalse(LinkTypes.isAnagram("cat", "cat"));
        assertTrue(LinkTypes.isAnagram("red sun", "nursed"));
        assertTrue(LinkTypes.isAnagram("nursed", "red sun"));
        assertTrue(LinkTypes.isAnagram("won't", "town"));
        assertTrue(LinkTypes.isAnagram("town", "won't"));
    }

    @Test
    public void testIsOneLetterChange() {
        assertTrue(LinkTypes.isOneLetterChange("moon", "moan"));
        assertTrue(LinkTypes.isOneLetterChange("cart", "care"));
        assertFalse(LinkTypes.isOneLetterChange("cat", "bit"));
        assertFalse(LinkTypes.isOneLetterChange("cat", "angle"));
        assertFalse(LinkTypes.isOneLetterChange("cat", "cat"));
    }

    @Test
    public void testIsOneLetterAdd() {
        assertTrue(LinkTypes.isOneLetterAdd("pan", "pain"));
        assertTrue(LinkTypes.isOneLetterAdd("car", "cart"));
        assertFalse(LinkTypes.isOneLetterAdd("cat", "bite"));
        assertFalse(LinkTypes.isOneLetterAdd("raw", "straw"));
        assertFalse(LinkTypes.isOneLetterAdd("cat", "cat"));
    }

    @Test
    public void testIsOneLetterRemove() {
        assertTrue(LinkTypes.isOneLetterRemove("pain", "pan"));
        assertTrue(LinkTypes.isOneLetterRemove("cart", "car"));
        assertFalse(LinkTypes.isOneLetterRemove("cat", "bite"));
        assertFalse(LinkTypes.isOneLetterRemove("straw", "raw"));
        assertFalse(LinkTypes.isOneLetterRemove("cat", "cat"));
    }

    @Test
    public void testToString1() {
        assertEquals("OneLetterChange", LinkType.OneLetterChange.toString());
        assertEquals("OneLetterAddOrRemove", LinkType.OneLetterAddOrRemove.toString());
        assertEquals("Anagram", LinkType.Anagram.toString());
        assertEquals("Synonym", LinkType.Synonym.toString());
        assertEquals("Antonym", LinkType.Antonym.toString());
        assertEquals("WordAssociation", LinkType.WordAssociation.toString());
    }
    
    @Test
    public void testCalculateLinkType() {
        assertEquals(LinkType.OneLetterChange, LinkTypes.calculateLinkType(cat, bat));
        assertEquals(LinkType.Anagram, LinkTypes.calculateLinkType(sport, ports));
        assertEquals(LinkType.OneLetterAddOrRemove, LinkTypes.calculateLinkType(port, ports));
        assertEquals(null, LinkTypes.calculateLinkType(bat, baseball));
    }

////        @Test
////        public void fromString1() {
////        assertEquals(LinkType.OneLetterChange, LinkTypes.fromString("OneLetterChange"));
////        assertEquals(LinkType.OneLetterAddOrRemove, LinkTypes.fromString("OneLetterAddOrRemove"));
////        assertEquals(LinkType.Anagram, LinkTypes.fromString("Anagram"));
////        assertEquals(LinkType.Synonym, LinkTypes.fromString("Synonym"));
////        assertEquals(LinkType.Antonym, LinkTypes.fromString("Antonym"));
////        assertEquals(LinkType.WordAssociation, LinkTypes.fromString("WordAssociation"));
////        Assert.ThrowsException<System.Exception>
////        (() => LinkTypes.fromString("Wrong"));
//        }

}
